core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -180,6 +211,54 @@ public String getDisplayName() {
         }
     }
 
+    private void createInitialApiToken(User user) throws IOException, InterruptedException {
+        ApiTokenProperty apiTokenProperty = user.getProperty(ApiTokenProperty.class);
+
+        String sysProp = ADMIN_INITIAL_API_TOKEN;
+        if (sysProp.equals("true")) {
+            TokenUuidAndPlainValue tokenUuidAndPlainValue = apiTokenProperty.generateNewToken("random-generation-during-setup-wizard");
+            FilePath fp = getInitialAdminApiTokenFile();
+            // same comment as in the init method
+
+            // JENKINS-33599 - write to a file in the jenkins home directory
+            // most native packages of Jenkins creates a machine user account 'jenkins' to run Jenkins,
+            // and use group 'jenkins' for admins. So we allow groups to read this file
+            fp.touch(System.currentTimeMillis());
+            fp.chmod(0640);
+            fp.write(tokenUuidAndPlainValue.plainValue, StandardCharsets.UTF_8.name());
+            LOGGER.log(Level.INFO, "The API Token was randomly generated and the information was put in {0}", fp.getRemote());
+        } else {
+            String plainText;
+            if (sysProp.startsWith("@")) {
+                // no need for path traversal check as it's coming from the instance creator only
+                File apiTokenFile = new File(sysProp.substring(1));
+                if (!apiTokenFile.exists()) {
+                    LOGGER.log(Level.WARNING, "The API Token cannot be retrieved from a inexistent file: {0}", apiTokenFile);
+                    return;
+                }
+
+                try {
+                    plainText = FileUtils.readFileToString(apiTokenFile, StandardCharsets.UTF_8);
+                    LOGGER.log(Level.INFO, "The API Token was generated using the file referenced by the given system property");
+                } catch (IOException e) {
+                    LOGGER.log(Level.WARNING, String.format("The API Token cannot be retrieved from the file: %s", apiTokenFile), e);
+                    return;
+                }
+            } else {
+                LOGGER.log(Level.INFO, "The API Token was generated using directly the given system property");
```

Comment:
should probably be a variable
```suggestion
                LOGGER.log(Level.INFO, "API Token generated using system property: " + SetupWizard.class.getName() + ".adminInitialApiToken");
```